control-plane/Dockerfile

```diff
@@ -92,7 +92,11 @@ LABEL name=${BIN_NAME} \
 ENV BIN_NAME=${BIN_NAME}
 ENV VERSION=${PRODUCT_VERSION}
 
-RUN apk add --no-cache ca-certificates libcap openssl su-exec iputils libc6-compat iptables
+RUN apk add --no-cache ca-certificates libcap openssl su-exec iputils gcompat libc6-compat libstdc++ iptables
+
+# for FIPS CGO glibc compatibility in alpine
+# see https://github.com/golang/go/issues/59305
+RUN ln -s /lib/libc.so.6 /usr/lib/libresolv.so.2
```

Comment:
What happens here when we upgrade to Go 1.21? Will this cause problems?

Reply:
I don't think Go is on track to re-include these C libraries going forward. (for instance: https://github.com/golang/go/issues/59305#issuecomment-1488478737)
This may not be required (i.e. could be removed) if Alpine decides to include this functionality in the future to address the issues that minikube and others are having: https://gitlab.alpinelinux.org/alpine/aports/-/issues/14846
Either way, this should not cause any problems.

An alternative to consider would be: https://honnef.co/articles/statically-compiled-go-programs-always-even-with-cgo-using-musl/
But that involves deeper changes and new build pipeline dependencies, which we can discuss for future releases.

Reply:
I can live with that